src/utils/TypeUtils.ts

```diff
@@ -0,0 +1,14 @@
+import { User } from "../sharedTypes";
+
+
+export class TypeUtils {
+    
+    public isUser(value: unknown): value is User {
```

Comment:
Don't try to make everything an object. You can keep these kind of stuff functions. It's not wrong to have functional code in your project.